### DISCUSSION TITLE
Freezing interface when pipeline server goes offline.

### DISCUSSION THREAD

**Discussion by aqualx** [open-webui/open-webui#5733] [General]
# Bug Report

## Installation Method
I have deployed Ollama+ Open WebUI + Pipiline docker images via this [docker-compose](https://github.com/aqualx/truenas-docker-compose/blob/master/apps/ollama/compose.yaml) template.
In addition to this, I'm running a development version of the [pipeline project](https://github.com/open-webui/pipelines) locally.
At Admin Panel -> Settings -> Connections-> I have two "OpenAI API" servers defined.

## Environment

- **Open WebUI Version:** v0.3.30
- **Ollama (if applicable):** latest as of 26.09.2024

- **Operating System:** Docker container running on TrueNas Scale 24.10
- **Browser (if applicable):** Vivaldi 6.9.3447.46

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
When one of "OpenAI API" service is offline I can:
  * use Open WebUI interface
  * change settings at Admin Panel > Settings. 
  * see some error messages that service is unavailable

## Actual Behavior:
When one of "OpenAI API" service is offline:
 * huge lags when accessing Admin Panel -> Settings -> Connection & Pipelines tabs
 * can't remove offline service from Admin Panel -> Settings -> Connection. When I try to delete it UI says "Settings saved successfully", but after reloading I see the old ones are in place.

## Description

**Bug Summary:**
Improper handling of offline OpenAI API service. Unavailable service should not freeze the UI as it gives customers an unreliable experience with a product.

**Browser Console Logs:**

> Socket undefined disconnected due to ping timeout
> 2+layout.svelte:54 connect_error Error: timeout
>     at manager.js:137:25
> index.ts:207 
>         
>         
>        GET https://my_host_name/openai/models/1 504 (Gateway Timeout)
> window.fetch @ fetcher.js:76
> y @ index.ts:207
> (anonymous) @ Connections.svelte:161
> (anonymous) @ Connections.svelte:160
> await in (anonymous)
> J @ utils.js:41
> (anonymous) @ Component.js:47
> _t @ scheduler.js:99
> Promise.then
> ut @ scheduler.js:20
> ht @ Component.js:81
> (anonymous) @ Component.js:139
> o @ Settings.svelte:100
> Show 7 more frames
> Show less
> Connections.svelte:163 Uncaught (in promise) OpenAI: Network Problem
> (anonymous) @ Connections.svelte:163
> await in (anonymous)
> (anonymous) @ Connections.svelte:160
> await in (anonymous)
> J @ utils.js:41
> (anonymous) @ Component.js:47
> _t @ scheduler.js:99
> Promise.then
> ut @ scheduler.js:20
> ht @ Component.js:81
> (anonymous) @ Component.js:139
> o @ Settings.svelte:100
> Show 6 more frames
> Show less
> index.ts:375 
>         
>         
>        GET https://my_host_name/api/pipelines/list 504 (Gateway Timeout)
> window.fetch @ fetcher.js:76
> A @ index.ts:375
> (anonymous) @ Pipelines.svelte:187
> J @ utils.js:41
> (anonymous) @ Component.js:47
> _t @ scheduler.js:99
> Promise.then
> ut @ scheduler.js:20
> ht @ Component.js:81
> (anonymous) @ Component.js:139
> o @ Settings.svelte:283
> Show 7 more frames
> Show less
> index.ts:388 SyntaxError: Unexpected token '<', "<html>
> <h"... is not valid JSON
> VM1214:1 Uncaught (in promise) SyntaxError: Unexpected token '<', "<html>
> <h"... is not valid JSON
> await
> J @ utils.js:41
> (anonymous) @ Component.js:47
> _t @ scheduler.js:99
> Promise.then
> ut @ scheduler.js:20
> ht @ Component.js:81
> (anonymous) @ Component.js:139
> o @ Settings.svelte:283
> Show 6 more frames
> Show less
> +layout.svelte:54 connect_error Error: timeout
>     at manager.js:137:25
> polling.js:298 
>         
>         
>        GET https://my_host_name/ws/socket.io/?EIO=4&transport=polling&t=P8kHgeD 504 (Gateway Timeout)
> create @ polling.js:298
> me @ polling.js:237
> request @ polling.js:190
> doPoll @ polling.js:215
> poll @ polling.js:96
> doOpen @ polling.js:56
> open @ transport.js:46
> open @ socket.js:170
> Oe @ socket.js:111
> open @ manager.js:108
> (anonymous) @ manager.js:328
> setTimeout
> reconnect @ manager.js:321
> onclose @ manager.js:301
> L.emit @ index.js:136
> onClose @ socket.js:570
> (anonymous) @ socket.js:377
> setTimeout
> resetPingTimeout @ socket.js:376
> onPacket @ socket.js:324
> L.emit @ index.js:136
> onPacket @ transport.js:98
> onData @ transport.js:90
> ws.onmessage @ websocket.js:68
> Show 22 more frames
> Show less
> +layout.svelte:54 connect_error Error: timeout

**Comment by tjbck** [comment]
Unable to reproduce, feel free to make a PR if you find the fix.

**Comment by jeremykei** [comment]
This also happens while the main event thread on server backend is an async pipe going heavy on some knowledge search (collection attached to model) and there's more than 2 or 3 documents. The client side timeout after about 15 seconds with the message "Searching Knowledge for -- insert query statement here -- ".

![image](https://github.com/user-attachments/assets/6c0090b4-592f-4917-a035-bc217cc844d1)

**Comment by jeremykei** [comment]
https://github.com/open-webui/open-webui/issues/6798

**Comment by jeremykei** [comment]
Tested to be fixed in current dev for case when socket timed out. Thanks!

![image](https://github.com/user-attachments/assets/beabbb6e-5e97-4d45-aa8f-9596ce0ff0cd)

**Comment by Roberto-Candelario** [comment]
where you able to find a fix for it? I cannot get. And by any chance are you using a GPU if so which one?
